inc/namespace.php

```diff
@@ -267,6 +271,21 @@ function add_login_logo() {
 	<?php
 }
 
+/**
+ * Filter to set the favicon url.
+ *
+ * @param string $url Filters the site icon URL.
+ * @return string $url The URL for the favicon.
+ */
+function filter_favicon( string $url ) {
+	if ( empty( $url ) ) {
+		$favicon = Altis\get_config()['modules']['cms']['favicon'];
+		$url = get_site_url( get_main_site_id( get_main_network_id() ), $favicon );
```

Comment:
Oh actually - this is a heavy function to run as it involves `switch_to_blog()` on any non primary site. We have this function in the Cloud module that could be good to put in the core module, for now though I think we just need to cache the value set from the config to avoid the `get_site_url()` call after the first run:

https://github.com/humanmade/altis-cloud/blob/7b04b086ab37bc2156647649c7f6a26d73ce9c51/inc/namespace.php#L541-L555